ludwig/features/base_feature.py

```diff
@@ -233,29 +217,27 @@ def update_metrics(self, targets: Tensor, predictions: Dict[str, Tensor]):
         for _, metric_fn in self.metric_functions.items():
             metric_class = type(metric_fn)
             prediction_key = metric_class.get_inputs()
+            # TODO(shreya): Metrics should ideally just move to the correct device
+            #  and not require the user to do this. This is a temporary fix. See
+            #  if this can be removed before merging the PR.
+            metric_fn = metric_fn.to(predictions[prediction_key].device)
             metric_fn.update(predictions[prediction_key].detach(), targets)
 
     def get_metrics(self):
         metric_vals = {}
         for metric_name, metric_onj in self.metric_functions.items():
             try:
-                metric_vals[metric_name] = metric_onj.compute(
-                ).detach().numpy().item()
+                metric_vals[metric_name] = metric_onj.compute().detach().cpu().numpy().item()
```

Comment:
This sequence `compute().detach().cpu().numpy().item()` appears both here and ecd.py. Does it make sense to put this in a shared function, i.e. `get_detached_cpu_metric(metric_obj)`?

Reply:
Done.